### PR TITLE
Remove Sorts.contents

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

**Comment by ppedrot** [comment]
What's the motivation for the patch?

**Comment by SkySkimmer** [comment]
One less type means it's a bit simpler.

**Review by ppedrot — APPROVED** [review]
Didn't check every single line in depth but LGTM.

File: engine/namegen.ml [hunk 1]
```diff
@@ -137,8 +137,9 @@ let lowercase_first_char id = (* First character of a constr *)
         s ^ Unicode.lowercase_first_char s'
 
 let sort_hdchar = function
-  | Prop(_) -> "P"
-  | Type(_) -> "T"
+  | Prop -> "P"
+  | Set -> "S"
+  | Type _ -> "T"
```
Comment (SkySkimmer) [inline engine/namegen.ml]: Here.

File: plugins/ltac/taccoerce.ml [hunk 2]
```diff
@@ -199,11 +199,12 @@ let id_of_name = function
...
        | Sort s ->
-	  begin
+          begin
 	    match ESorts.kind sigma s with
-	    | Sorts.Prop _ -> Label.to_id (Label.make "Prop")
-	    | Sorts.Type _ -> Label.to_id (Label.make "Type")
-	  end
+            | Sorts.Prop -> Label.to_id (Label.make "Prop")
+            | Sorts.Set -> Label.to_id (Label.make "Set")
+            | Sorts.Type _ -> Label.to_id (Label.make "Type")
```
Comment (SkySkimmer) [inline plugins/ltac/taccoerce.ml]: Here.

**Review by SkySkimmer — COMMENTED** [review]
(There are a couple changes in generated strings, see comments.)

**Comment by ppedrot** [comment]
@SkySkimmer Probably rare, almost nobody uses Set explicity.

**Comment by ppedrot** [comment]
@SkySkimmer  Can you rebase and fix elpi?

**Review by maximedenes — APPROVED** [review]
I like the change. I haven't checked if it was correctly propagated everywhere, but I trust you :)

**Comment by SkySkimmer** [comment]
See https://github.com/LPCIC/coq-elpi/pull/27

**Comment by ppedrot** [comment]
@mattam82 Are you fine with this change?

@SkySkimmer You should rebase.

**Comment by ppedrot** [comment]
@mattam82 is overloaded and this is trivial enough that if I don't get an answer in a few days, I'll merge it.

**Pull request merged**